File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/inline/comment/direct/Java/C/s196249385_commented.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_N 1000

int compare(const void *a, const void *b) {
    return (*(long *)b - *(long *)a);
}

void solve() {
    int n, k;
    scanf("%d %d", &n, &k);

    int a[MAX_N];
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }

    long cumsum[MAX_N + 1];
    cumsum[0] = 0;
    for (int i = 0; i < n; i++) {
        cumsum[i + 1] = cumsum[i] + a[i];
    }

    long b[MAX_N * (MAX_N + 1) / 2];
    int size_b = 0;
    long max = 0;
    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {
            b[size_b++] = cumsum[j + 1] - cumsum[i];
            if (cumsum[j + 1] - cumsum[i] > max) {
                max = cumsum[j + 1] - cumsum[i];
            }
        }
    }

    long mask = 1;
    while (mask <= max) {
        mask <<= 1;
    }
    mask >>= 1;

    while (mask > 0) {
        long tmp[MAX_N * (MAX_N + 1) / 2];
        int size_tmp = 0;
        for (int i = 0; i < size_b; i++) {
            if (b[i] & mask) {
                tmp[size_tmp++] = b[i];
            }
        }

        if (size_tmp >= k) {
            size_b = size_tmp;
            for (int i = 0; i < size_b; i++) {
                b[i] = tmp[i];
            }
        }

        mask >>= 1;
    }

    if (size_b < k) {
        printf("0\n");
    } else {
        long ans = -1L;
        for (int i = 0; i < size_b; i++) {
            ans &= b[i];
        }

        printf("%ld\n", ans);
    }
}

int main() {
    solve();
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/uncomment/direct/Java/Go/s196249385_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func solve(sc *bufio.Reader, pr *bufio.Writer) {
	var n, k int
	fmt.Fscan(sc, &n, &k)

	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscan(sc, &a[i])
	}

	cumsum := make([]int, n+1)
	for i := 0; i < n; i++ {
		cumsum[i+1] = cumsum[i] + a[i]
	}

	b := []int{}
	max := 0
	for i := 0; i < n; i++ {
		for j := i; j < n; j++ {
			diff := cumsum[j+1] - cumsum[i]
			b = append(b, diff)
			if diff > max {
				max = diff
			}
		}
	}

	mask := 1 << uint(bitCount(max))
	for mask > 0 {
		tmp := []int{}
		for _, e := range b {
			if e&mask != 0 {
				tmp = append(tmp, e)
			}
		}

		if len(tmp) >= k {
			b = tmp
		}

		mask >>= 1
	}

	if len(b) < k {
		fmt.Fprintln(pr, 0)
	} else {
		ans := -1
		for _, e := range b {
			ans &= e
		}

		fmt.Fprintln(pr, ans)
	}
}

func bitCount(n int) int {
	count := 0
	for n > 0 {
		count++
		n &= n - 1
	}
	return count
}

func main() {
	sc := bufio.NewReader(os.Stdin)
	pr := bufio.NewWriter(os.Stdout)
	defer pr.Flush()

	solve(sc, pr)
}

// End of Code
